src/mount/readdata.cc

```diff
@@ -663,15 +672,15 @@ int read_to_buffer(ReadRecord *rrec, uint64_t current_offset,
 			uint32_t read_cache_bytes_to_reserve =
 				getBytesToBeReadFromCS(reader.index(), offset_in_chunk,
 										size_in_chunk, reader.fileLength());
-			std::unique_lock lock(gReadCacheMemoryMutex);
+			std::unique_lock usedMemoryLock(gReadCacheMemoryMutex);
```

Comment:
Consider if it is optimal creating a deferred version of this lock outside the container `while` and then lock and unlock as needed. Consider extending the change to all `usedMemoryLock` uses inside the upper `while`.

Reply:
Let's add this change in further PRs, that we are now currently working on. Thank you.